ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/services/DefaultKsqlClient.java

```diff
@@ -125,14 +125,19 @@ public void makeAsyncLagReportRequest(
     getTarget(target, authHeader).postAsyncLagReportingRequest(lagReportingMessage);
   }
 
+  @Override
+  public void close() {
+    sharedClient.close();
+  }
+
   private KsqlTarget getTarget(final KsqlTarget target, final Optional<String> authHeader) {
     return authHeader
         .map(target::authorizationHeader)
         .orElse(target);
   }
 
   private static HttpClientOptions createClientOptions() {
-    return new HttpClientOptions().setMaxPoolSize(100);
```

Comment:
Out of curiosity, how come this was removed?

Reply:
My reasoning was, if the client is being created for each request, then no point in having a large pool size. But perhaps this is not happening??